source/common/secret/secret_manager_impl.cc

```diff
@@ -79,5 +84,57 @@ SecretManagerImpl::findOrCreateCertificateValidationContextProvider(
                                                     secret_provider_context);
 }
 
+// TODO: question, what's the handling of static inlined stuff? they're just internal constructs
+// not needed to be exposed, maybe?
+ProtobufTypes::MessagePtr SecretManagerImpl::dumpSecretConfigs() {
+  auto config_dump = std::make_unique<envoy::admin::v2alpha::SecretsConfigDump>();
+  auto providers = certificate_providers_.allSecretProviders();
+  for (const auto& cert_secrets : providers) {
+    const auto& secret_data = cert_secrets->secretData();
+    const auto& tls_cert = cert_secrets->secret();
+    const auto& dynamic_secret = config_dump->mutable_dynamic_secrets()->Add();
+    const auto& secret = dynamic_secret->mutable_secret();
+
+    ProtobufWkt::Timestamp last_updated_ts;
+    TimestampUtil::systemClockToTimestamp(secret_data.last_updated_, last_updated_ts);
+    dynamic_secret->set_version_info(secret_data.version_info_);
+    *dynamic_secret->mutable_last_updated() = last_updated_ts;
+    secret->set_name(secret_data.resource_name);
+    // TODO(incfly): this means the config is registered but Envoy hasn't received a
```

Comment:
After getting more familiar with other parts, I now think we can have a `warming_secrets` in the config dump similar to listeners config dump, and then put cases like this to the list.

let me know if that makes sense.

Reply:
that sounds right.

Reply:
Done.